ambry-replication/src/test/java/com.github.ambry.replication/CloudToStoreReplicationManagerTest.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.replication;
+
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.clustermap.MockClusterMap;
+import com.github.ambry.clustermap.MockClusterSpectator;
+import com.github.ambry.clustermap.MockDataNodeId;
+import com.github.ambry.clustermap.MockHelixParticipant;
+import com.github.ambry.clustermap.MockPartitionId;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.DiskManagerConfig;
+import com.github.ambry.config.ReplicationConfig;
+import com.github.ambry.config.ServerConfig;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.network.Port;
+import com.github.ambry.network.PortType;
+import com.github.ambry.store.MockStoreKeyConverterFactory;
+import com.github.ambry.store.StorageManager;
+import com.github.ambry.store.StoreKey;
+import com.github.ambry.store.StoreKeyFactory;
+import com.github.ambry.utils.MockTime;
+import com.github.ambry.utils.TestUtils;
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ScheduledExecutorService;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static com.github.ambry.clustermap.CloudReplica.*;
+import static com.github.ambry.clustermap.TestUtils.*;
+import static org.junit.Assert.*;
+
+
+/**
+ * Tests for {@link CloudToStoreReplicationManager} when adding/removing cloud replica. The tests also cover the case
+ * where new replica is added due to "move replica".
+ */
+public class CloudToStoreReplicationManagerTest {
+  private static final String NEW_PARTITION_NAME = "12";
+  private static final String CLOUD_DC_NAME = "CloudDc";
+  private static final String VCR_MOUNT_PATH = "/vcr/1";
+  private static final String VCR_REPLICA_THREAD_PREFIX = "VcrReplicaThread-";
```

Comment:
I was wondering where this hardcoded constant was coming from and found this horrible line of code in ReplicationEngine.createThreadPool():
      String threadIdentity =  (startThread ? "Vcr" : "") + "ReplicaThread-" + ...
I'm adding a TODO for myself to fix that and this constant as well.  It's okay for now.

Reply:
yeah, we probably need to fix it in ReplicationEngine as well.